### DISCUSSION TITLE
Unable to load preload

### DISCUSSION THREAD

**Discussion by nadaa** [electron-vite/electron-vite-vue#383] [Q&A]
Hi,
I keep getting an error when I add contextbridge in the preload.
 Unable to load preload script: \electron-vite-project\dist-electron\preload\index.js
node:electron/js2c/renderer_init:2 Error: contextBridge API can only be used when contextIsolation is enabled.

I tried to enable **contextisolation** from the main process, but did not work.
Any suggestion on how to resolve this issue. I have Electron **24.1.3**

Thanks

**Comment by caoxiemeihao** [comment]
- electron/main/index.ts
```diff, ts
  win = new BrowserWindow({
    title: 'Main window',
    icon: join(process.env.PUBLIC, 'favicon.ico'),
    webPreferences: {
      preload,
      // Warning: Enable nodeIntegration and disable contextIsolation is not secure in production
      // Consider using contextBridge.exposeInMainWorld
      // Read more on https://www.electronjs.org/docs/latest/tutorial/context-isolation
-     nodeIntegration: true,
-     contextIsolation: false,
    },
  })
```

**Reply by nadaa** [reply]
Thanks. I tried it but it is not working.

**Reply by caoxiemeihao** [reply]
Maybe the [electron-vite-boilerplate](https://github.com/electron-vite/electron-vite-boilerplate) is better for you :)

**Reply by nadaa** [reply]
Ok, I will try. 
I have tried many installations :)
